cli/module_generate.go

```diff
@@ -0,0 +1,726 @@
+			return
+		}
+
+		s.Title("Rendering common files...")
+		if err = renderCommonFiles(cCtx, *newModule); err != nil {
+			fatalError = err
+			return
+		}
+
+		s.Title(fmt.Sprintf("Copying %s files...", newModule.Language))
+		if err = copyLanguageTemplate(cCtx, newModule.Language, newModule.ModuleName); err != nil {
+			fatalError = err
+			return
+		}
+
+		s.Title("Rendering template...")
+		if err = renderTemplate(cCtx, *newModule); err != nil {
+			fatalError = err
+			return
+		}
+
+		s.Title(fmt.Sprintf("Generating %s stubs...", newModule.Language))
+		if err = generateStubs(cCtx, *newModule); err != nil {
+			warningf(cCtx.App.ErrWriter, err.Error())
+			nonFatalError = true
+		}
+
+		s.Title("Generating cloud build requirements...")
+		if err = generateCloudBuild(cCtx, *newModule); err != nil {
+			warningf(cCtx.App.ErrWriter, err.Error())
+			nonFatalError = true
+		}
+	}
+
+	if cCtx.Bool(debugFlag) {
+		action()
+	} else {
+		s.Action(action)
+		err := s.Run()
+		if err != nil {
+			return err
+		}
+	}
+
+	if fatalError != nil {
+		err := os.RemoveAll(newModule.ModuleName)
+		if err != nil {
+			return errors.Wrap(fatalError, fmt.Sprintf("some steps of module generation failed, "+
+				"incomplete module located at %s", newModule.ModuleName))
+		}
+		return errors.Wrap(fatalError, "unable to generate module")
+	}
+
+	if nonFatalError {
+		return fmt.Errorf("some steps of module generation failed, incomplete module located at %s", newModule.ModuleName)
+	}
+
+	printf(cCtx.App.Writer, "Module successfully generated at %s", newModule.ModuleName)
+	return nil
+}
+
+// Prompt the user for information regarding the module they want to create
+// returns the moduleInputs struct that contains the information the user entered.
+func promptUser() (*moduleInputs, error) {
+	var newModule moduleInputs
+	form := huh.NewForm(
+		huh.NewGroup(
+			huh.NewNote().
+				Title("Generate a new modular resource").
+				Description("For more details about modular resources, view the documentation at \nhttps://docs.viam.com/registry/"),
+			huh.NewInput().
+				Title("Set a module name:").
+				Description("The module name can contain only alphanumeric characters, dashes, and underscores.").
+				Value(&newModule.ModuleName).
+				Placeholder("my-module").
+				Suggestions([]string{"my-module"}).
+				Validate(func(s string) error {
+					if s == "" {
+						return errors.New("module name must not be empty")
+					}
+					match, err := regexp.MatchString("^[a-z0-9]+(?:[_-][a-z0-9]+)*$", s)
+					if !match || err != nil {
+						return errors.New("module names can only contain alphanumeric characters, dashes, and underscores")
+					}
+					if _, err := os.Stat(s); err == nil {
+						return errors.New("this module directory already exists")
+					}
+					return nil
+				}),
+			huh.NewSelect[string]().
+				Title("Specify the language for the module:").
+				Options(
+					huh.NewOption("Python", python),
+					// huh.NewOption("Go", "go"),
```

Comment:
I agree, but want to keep this in and not change it because i know there's work going on in golang now that would uncomment this line, and that would make for an annoying merge commit since both of these lines would be changed :( but good to note and I agree for future